src/v/cloud_storage/segment_chunk_api.cc

```diff
@@ -149,10 +150,64 @@ ss::future<segment_chunk::handle_t> segment_chunks::hydrate_chunk(
       chunk_start,
       chunk.waiters.size());
 
+    auto prefetch = prefetch_override.value_or(
+      config::shard_local_cfg().cloud_storage_chunk_prefetch);
+    schedule_prefetches(chunk_start, prefetch);
+
     _bg_cvar.signal();
     co_return co_await std::move(f);
 }
 
+void segment_chunks::schedule_prefetches(
+  chunk_start_offset_t start_offset, size_t prefetch) {
+    for (auto it = std::next(_chunks.find(start_offset));
+         it != _chunks.end() && prefetch > 0;
+         ++it, --prefetch) {
+        auto& [start, chunk] = *it;
+        if (
+          chunk.current_state == chunk_state::download_in_progress
+          || chunk.current_state == chunk_state::hydrated) {
+            continue;
+        }
+
+        chunk.current_state = chunk_state::download_in_progress;
+        ss::promise<cloud_storage::segment_chunk::handle_t> p;
+        _prefetches.emplace_back(p.get_future());
+        chunk.waiters.push_back(
+          std::move(p), ss::lowres_clock::time_point::max());
+        vlog(
+          _ctxlog.trace,
+          "prefetch hydrate request added for chunk id {}, current "
+          "waiter size: {}",
+          start,
+          chunk.waiters.size());
+    }
+}
+
+void segment_chunks::resolve_prefetch_futures() {
+    auto available_it = std::ranges::partition(
+      _prefetches, [](const auto& f) { return !f.available(); });
+
+    fragmented_vector<ss::future<segment_chunk::handle_t>> resolved;
+    resolved.reserve(available_it.size());
+
+    std::ranges::move(available_it, std::back_inserter(resolved));
+
+    if (!resolved.empty()) {
+        vlog(_ctxlog.trace, "{} completed prefetches", resolved.size());
+        for (auto& f : resolved) {
+            vassert(
+              f.available(), "future not available when resolving prefetches");
+            if (f.failed()) {
+                vlog(
+                  _ctxlog.trace,
+                  "failed prefetch download: {}",
```

Comment:
I'm curious how common we expect these to be. Should they be logged at warning or info level? Are they helpful to know about?

Reply:
Since we do not hold the gate for prefetch operations, I expect these to be mostly gate closed exceptions when the segment is shut down, originating from the waiter cleanup code discussed in this thread https://github.com/redpanda-data/redpanda/pull/18278#discussion_r1600061917, which can be ignored. 

If there are real HTTP errors we should get warnings from the http client etc.